jwst/resample/resample_utils.py

```diff
@@ -163,11 +163,18 @@ def reproject(wcs1, wcs2):
     """
 
     try:
+        # Here we want to use the WCS API functions so that a Sliced WCS
+        # will work as well. However, the API functions do not accept
+        # keyword arguments and `with_bounding_box=False` cannot be passsed.
```

Comment:
what would it take to make the relevant function accept the `with_bounding_box=False` argument?

Reply:
The question has been raised with other developers in the community and the answer is always that there shouldn't be kwargs passed to these fnuctions. Part of the reason is that the shared API should work with other WCS libraries and the bounding_box is gwcs specific.

Reply:
What harm would occur if shared API would have `with_bounding_box` but it would have no effect on, e.g., FITS WCS? Especially if the default is `with_bounding_box=False`. Then only users who explicitly need this feature will set `with_bounding_box=True` (with no effect on FITS WCS).

Reply:
What harm would occur if shared API would have with_bounding_box but it would have no effect on, e.g., FITS WCS? Especially if the default is with_bounding_box=False. Then only users who explicitly need this feature will set with_bounding_box=True (with no effect on FITS WCS).